core/sail/base/src/main/java/org/eclipse/rdf4j/sail/base/Changeset.java

```diff
@@ -833,8 +847,9 @@ void sinkApproved(SailSink sink) {
 
 		boolean readLock = readWriteLock.readLock();
 		try {
-			if (approved != null) {
-				sink.approveAll(approved, approvedContexts);
+			Model approvedLocal = approved;
+			if (approvedLocal != null) {
+				sink.approveAll(approvedLocal, approvedContexts);
```

Comment:
I had assumed that this already was safe since it's wrapped in a read lock.

Reply:
I was not sure if it interferes with `close` as the latter does not use the lock while changing the variables.

Reply:
I don't think any of our code tries to read or write concurrently with close(). Or at least I hope not :-/